great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -819,14 +819,26 @@ def qualified_name(self) -> str:
     @pydantic.validator("table_name", pre=True, always=True)
     def _default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
         if not (validated_table_name := table_name or values.get("name")):
+            # TODO: if using `name` we should replace spaces with underscores
+            # (unless bracketed by quotes)
             raise ValueError(
                 "table_name cannot be empty and should default to name if not provided"
             )
 
+        table_name_is_qouted: bool = cls._is_bracketed_by_quotes(validated_table_name)
+
         from great_expectations.compatibility import sqlalchemy
 
         if sqlalchemy.quoted_name:
-            return sqlalchemy.quoted_name(value=validated_table_name, quote=True)
+            if table_name_is_qouted:
+                # Remove the quotes and add them back using the sqlalchemy.quoted_name function
+                # TODO: do we need to worry about nested qoutes?
+                validated_table_name = validated_table_name.strip("'").strip('"')
+
+            return sqlalchemy.quoted_name(
+                value=validated_table_name,
+                quote=table_name_is_qouted,
+            )
```

Comment:
This feels a little silly.
I don't totally understand what problem the original `sqlalchemy.quoted_name` was solving.
But I believe this solution preserves the intended fix this was meant to apply.

Reply:
This logic will be moved to the specific `SQLDatasource` subclasses that require it.
Example (SnowflakeDatasource)